lib/Mapper.ts

```diff
@@ -52,7 +52,9 @@ export default class Mapper {
 
   async *files(): AsyncIterableIterator<File> {
     const { options } = this.parsed;
-    yield* this.parsed.fileNames.map(path => new EsFile({ path, options, config: this.parsed }));
+    yield* this.parsed.fileNames
+      .filter(path => options.declaration || !path.endsWith('.d.ts'))
+      .map(path => new EsFile({ path, options, config: this.parsed }));
     if (options.declaration) {
       yield* this.parsed.fileNames.map(path => new TsFile({ path, options, config: this.parsed }));
```

Comment:
We need to filter the files in both locations so the following code would do that:

```ts
  async *files(): AsyncIterableIterator<File> {
    const parsed = this.parsed;
    const { options, fileNames } = parsed;
    const files = fileNames.filter(n => !n.endsWith('.d.ts'));
    yield* files.map(path => new EsFile({ path, options, config: this.parsed }));
    if (options.declaration) {
      yield* files.map(path => new TsFile({ path, options, config: this.parsed }));
    }
  }
```

Thanks for the fix 👍